homeassistant/components/onvif/parsers.py

```diff
@@ -397,33 +403,57 @@ async def async_parse_tplink_detector(uid: str, msg) -> Event | None:
 
         for item in payload.Data.SimpleItem:
             if item.Name == "IsVehicle":
```

Comment:
This is starting to get a bit long.  I'd change it to use a dict lookup or dispatch table instead

Reply:
How's this look?